dbt/contracts/graph/compiled.py

```diff
@@ -1,48 +1,93 @@
-from voluptuous import Schema, Required, All, Any, Length
-
-from collections import OrderedDict
-
-from dbt.compat import basestring
-from dbt.exceptions import ValidationException
+from dbt.api import APIObject
 from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.utils import deep_merge
 
-from dbt.contracts.common import validate_with
-from dbt.contracts.graph.parsed import parsed_node_contract, \
-    parsed_macro_contract
-
-
-compiled_node_contract = parsed_node_contract.extend({
-    # compiled fields
-    Required('compiled'): bool,
-    Required('compiled_sql'): Any(basestring, None),
+from dbt.contracts.graph.parsed import PARSED_NODE_CONTRACT, \
+    PARSED_MACRO_CONTRACT
 
-    # injected fields
-    Required('extra_ctes_injected'): bool,
-    Required('extra_ctes'): All(OrderedDict, {
-        basestring: Any(basestring, None)
-    }),
-    Required('injected_sql'): Any(basestring, None),
-})
+COMPILED_NODE_CONTRACT = deep_merge(
+    PARSED_NODE_CONTRACT,
+    {
+        # TODO: when we add 'extra_ctes' back in, flip this back to False
```

Comment:
what's the deal with `extra_ctes`?

Reply:
They're stored as an `OrderedDict`, which doesn't have any nice JSON representation (the standard technique I've seen is to use a list of dicts like `[{'key': 'foo', 'value': 'bar'}, {'key': 'baz', 'value': 'quux'}, ...]`). So to store it as schema-validated JSON, we'd end up spending a lot of time packing/unpacking everything into/from the `OrderedDict`/list-of-dicts.

A better longer-term path is probably to convert the `extra_ctes` into just an array, or leave it out of the contract, like I did with the agate tables. I briefly tested out the array change and broke a lot of tests, so obviously there's something I don't understand going on.